libminifi/include/core/repository/VolatileRepository.h

```diff
@@ -280,6 +290,19 @@ bool VolatileRepository<T>::Put(T key, const uint8_t *buf, size_t bufLen) {
   logger_->log_debug("VolatileRepository -- put %u %u", current_size_.load(), current_index_.load());
   return true;
 }
+
+template<typename T>
+bool VolatileRepository<T>::MultiPut(const std::vector<std::pair<T, std::unique_ptr<io::DataStream>>>& data) {
+  bool success = true;
+  for (const auto& item : data) {
+    success &= Put(item.first, item.second->getBuffer(), item.second->getSize());
```

Comment:
This does not result in a transacted MultiPut, but I am not sure how that can be achieved with the current architecture of the VolatileRepositories. A rethink of the entire implementation is definitely needed.

Reply:
That would still not be in a transaction. An another thread might have updated or deleted and reinserted one of those keys, which we would be deleting with your suggested method.

Reply:
Okay, that's fair.